src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -300,27 +302,24 @@ class PostgresEntityDataQueryService(
         val partitions = partitionsInfo.partitions.toList()
 
         entities.entries
-                .groupBy({ getPartition(it.key, partitions) }, { it.toPair() })
-                .mapValues { it.value.toMap() }
-                .asSequence().asStream().parallel()
-                .forEach { (partition, rawEntityBatch) ->
-
-                    val entityBatch = rawEntityBatch.mapValues { (entityKeyId, rawValue) ->
-                        return@mapValues if (awsPassthrough) {
-                            rawValue
+                .groupBy { getPartition(it.key, partitions) }
+                .forEach { (partition, batch) ->
+                    val entityBatch = batch.map { (entityKeyId, rawValue) ->
+                        return@map if (awsPassthrough) {
+                            entityKeyId to rawValue
                         } else {
-                            Multimaps.asMap(JsonDeserializer
+                            entityKeyId to Multimaps.asMap(JsonDeserializer
                                                     .validateFormatAndNormalize(rawValue, authorizedPropertyTypes)
                                                     { "Entity set $entitySetId with entity key id $entityKeyId" })
                         }
-                    }
+                    }.toMap()
 
                     tombstoneFn(version, entityBatch)
                     val upc = upsertEntities(
                             entitySetId,
                             entityBatch,
                             authorizedPropertyTypes,
-                            version,
+                            version+1,
```

Comment:
🤔

Reply:
I too am confused about this. What is this for?